libs/gl-client/src/credentials.rs

```diff
@@ -211,12 +230,34 @@ impl Device {
     pub fn to_bytes(&self) -> Vec<u8> {
         self.to_owned().into()
     }
+
+    pub fn node_id(&self) -> anyhow::Result<Vec<u8>> {
+        get_node_id_from_tls_config(&self.tls_config())
+    }
```

Comment:
Instead of deriving the `node_id` from deep within the `tls_config`, we could as well consider to add the `node_id` to the device credentials metadata (add it as a field in `Device`).